### ISSUE TITLE
Building behind a proxy

### ISSUE THREAD

**Issue by silverdaz** [broadinstitute/picard#641]
I'm sitting behind a proxy and I'm trying to build Picard.

When I input `./gradlew shadowJar`, I get the following error

```
Downloading` https://services.gradle.org/distributions/gradle-2.13-bin.zip

Exception in thread "main" java.net.UnknownHostException: services.gradle.org
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:668)
	at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:432)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:527)
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1105)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:999)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1513)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1441)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.gradle.wrapper.Download.downloadInternal(Download.java:58)
	at org.gradle.wrapper.Download.download(Download.java:44)
	at org.gradle.wrapper.Install$1.call(Install.java:61)
	at org.gradle.wrapper.Install$1.call(Install.java:48)
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:65)
	at org.gradle.wrapper.Install.createDist(Install.java:48)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
```

so I wanted to adjust the proxy settings.

I tried to pass them as arguments like
`./gradlew -Dhttp.proxyHost=AAA.BBB.CCC.DDD -Dhttp.proxyPort=PPPP shadowJar`

I tried to add them to `settings.gradle`
```
systemProp.http.proxyHost=AAA.BBB.CCC.DDD
systemProp.http.proxyPort=PPPP
```
and also to `./gradle/wrapper/gradle-wrapper.properties`

But I still get the same error that it cannot access https://services.gradle.org/distributions/gradle-2.13-bin.zip

**Comment by lbergelson** [comment]
@silverdaz Sorry for the problem and thanks for bringing it to our attention.  

Unfortunately I don't think we have any experience dealing with the gradle proxy settings.  The best way to get help with that is probably to ask on the gradle forums directly https://discuss.gradle.org/. 

In the interim I can suggest a few possible work arounds.
1.  Install gradle 2.13 manually.  Then you can use `gradle` directly instead of `./gradlew` in all of the build commands.  This would avoid problems with connecting to `services.gradle.org`, but you may still encounter problems downloading dependencies.  You might have to populate your maven cache manually as well which would be a bit of pain.
2.  Can you use the [prebuilt jar](https://github.com/broadinstitute/picard/releases/download/2.6.0/picard.jar)?  If you don't need to customize picad you could just download that.
3.  This ones a long shot, but I might try switching the link in the gradle-wrapper.properties from `https` -> `http`.  I've seen issues with https when running with something other than the complete oracle jvm.  Some jvms don't come with all the encryption libraries that the Oracle jvm does and will fail to make encrypted connections with certain servers that have disabled falling back to low security encryption.

Please let me know if you any of those solutions work for you or if you need additional help.

**Comment by silverdaz** [comment]
Ok, I managed. Here is what I did.

First, I installed gradle itself.
It was still giving me errors, because Java was trying to resolve the proxy address, which was of the form of an IP, AAA.BBB.CCC.DDD.
I am without internet, behind a proxy, so I thought it was some sort of catch 22 to require name resolution. I look around for some sort of '--nodns' option, but in vain.
Nevertheless, my solution was to add `AAA.BBB.CCC.DDD some_proxy_name` to `/etc/hosts` and use some_proxy_name as the proxy setting.

Moreover, I added the proxy settings to `~/.gradle/gradle.properties`
```
systemProp.http.proxyHost=some_proxy_name
systemProp.http.proxyPort=3128
```
To be on the safe side, I also used the `_JAVA_OPTIONS`
With the --info and --debug options, I could see that it picked up the variables from the _JAVA_OPTIONS environment variable. So I was safe and it worked.
